src/utils/iso8601.ts

```diff
@@ -30,17 +30,17 @@ export function iso8601DurationsToSeconds(iso8601: string): number {
             let i = 2;
 
             totalSecond = 0;
-            totalSecond += data[i] ? parseInt(data[i], 10) * year : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) * year : 0;
             i += 2;
-            totalSecond += data[i] ? parseInt(data[i], 10) * month : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) * month : 0;
             i += 2;
-            totalSecond += data[i] ? parseInt(data[i], 10) * day : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) * day : 0;
             i += 2;
-            totalSecond += data[i] ? parseInt(data[i], 10) * hour : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) * hour : 0;
             i += 2;
-            totalSecond += data[i] ? parseInt(data[i], 10) * minute : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) * minute : 0;
             i += 2;
-            totalSecond += data[i] ? parseInt(data[i], 10) : 0;
+            totalSecond += data[i] ? parseFloat(data[i]) : 0;
```

Comment:
As explained in my comment above about the `formatTime_()` function, I would strongly suggest extracting `nSeconds` `[00, 59]` and `nMilliseconds` `[000, 999]` to pass as integer function arguments. This will ensure consistency with the original function contract (which admittedly, is underspecified because of the generic `number` type which accepts floats and is non-bounded)

Reply:
I've disabled milliseconds .. Who one take care milliseconds on audiobooks ?

Reply:
ok.